rust_book_ex_8_3/src/hashmap_ex.rs

```diff
@@ -0,0 +1,32 @@
+use std::collections::HashMap;
+
+pub struct CompanyEmployees {
+    dep_employee_map: HashMap<String, Vec<String>>,
+}
+
+impl CompanyEmployees {
+    pub fn new() -> CompanyEmployees {
+        CompanyEmployees {
+            dep_employee_map: HashMap::new()
+        }
+    }
+
+    pub fn add_entry(&mut self, input: String) {
+        let splits = input.split(' ');
+        let tokens: Vec<&str> = splits.collect();
```

Comment:
First, you could omit `&str` here, like this (type inference will figure it out):

```rust
let tokens: Vec<_> = splits.collect();
```

Second, why do you need to collect this iterator into a vector here? (that's an extra allocation)